### PR TITLE
Update tool XSD for (hopefully deprecated) change_format syntax.

### LINKED ISSUE #3552: XSD for output data/when tag unaware of legacy attributes
I just ran into a planemo XSD linting failure in a very old tool, https://github.com/peterjc/pico_galaxy/commit/dff56b341e60347cbc6d9dbe6c8d89eacac61af6

```
$ planemo --version
planemo, version 0.38.0.dev0
$ git clone https://github.com/peterjc/pico_galaxy.git
$ cd pico_galaxy
$ git checkout dff56b341e60347cbc6d9dbe6c8d89eacac61af6
$ planemo lint tools/fastq/fastq_filter_by_id.xml
...
Applying linter tool_xsd... FAIL
.. ERROR: Invalid tmpKBNiWQ found. Errors [/tmp/tmpKBNiWQ:35:0:ERROR:SCHEMASV:SCHEMAV_CVC_COMPLEX_TYPE_3_2_1: Element 'when', attribute 'attribute': The attribute 'attribute' is not allowed.
/tmp/tmpKBNiWQ:35:0:ERROR:SCHEMASV:SCHEMAV_CVC_COMPLEX_TYPE_3_2_1: Element 'when', attribute 'input_dataset': The attribute 'input_dataset' is not allowed.
/tmp/tmpKBNiWQ:35:0:ERROR:SCHEMASV:SCHEMAV_CVC_COMPLEX_TYPE_4: Element 'when': The attribute 'input' is required but missing.
/tmp/tmpKBNiWQ:36:0:ERROR:SCHEMASV:SCHEMAV_CVC_COMPLEX_TYPE_3_2_1: Element 'when', attribute 'attribute': The attribute 'attribute' is not allowed.
/tmp/tmpKBNiWQ:36:0:ERROR:SCHEMASV:SCHEMAV_CVC_COMPLEX_TYPE_3_2_1: Element 'when', attribute 'input_dataset': The attribute 'input_dataset' is not allowed.
/tmp/tmpKBNiWQ:36:0:ERROR:SCHEMASV:SCHEMAV_CVC_COMPLEX_TYPE_4: Element 'when': The attribute 'input' is required but missing.
/tmp/tmpKBNiWQ:37:0:ERROR:SCHEMASV:SCHEMAV_CVC_COMPLEX_TYPE_3_2_1: Element 'when', attribute 'attribute': The attribute 'attribute' is not allowed.
/tmp/tmpKBNiWQ:37:0:ERROR:SCHEMASV:SCHEMAV_CVC_COMPLEX_TYPE_3_2_1: Element 'when', attribute 'input_dataset': The attribute 'input_dataset' is not allowed.
/tmp/tmpKBNiWQ:37:0:ERROR:SCHEMASV:SCHEMAV_CVC_COMPLEX_TYPE_4: Element 'when': The attribute 'input' is required but missing.
/tmp/tmpKBNiWQ:38:0:ERROR:SCHEMASV:SCHEMAV_CVC_COMPLEX_TYPE_3_2_1: Element 'when', attribute 'attribute': The attribute 'attribute' is not allowed.
/tmp/tmpKBNiWQ:38:0:ERROR:SCHEMASV:SCHEMAV_CVC_COMPLEX_TYPE_3_2_1: Element 'when', attribute 'input_dataset': The attribute 'input_dataset' is not allowed.
/tmp/tmpKBNiWQ:38:0:ERROR:SCHEMASV:SCHEMAV_CVC_COMPLEX_TYPE_4: Element 'when': The attribute 'input' is required but missing.
/tmp/tmpKBNiWQ:45:0:ERROR:SCHEMASV:SCHEMAV_CVC_COMPLEX_TYPE_3_2_1: Element 'when', attribute 'attribute': The attribute 'attribute' is not allowed.
/tmp/tmpKBNiWQ:45:0:ERROR:SCHEMASV:SCHEMAV_CVC_COMPLEX_TYPE_3_2_1: Element 'when', attribute 'input_dataset': The attribute 'input_dataset' is not allowed.
/tmp/tmpKBNiWQ:45:0:ERROR:SCHEMASV:SCHEMAV_CVC_COMPLEX_TYPE_4: Element 'when': The attribute 'input' is required but missing.
/tmp/tmpKBNiWQ:46:0:ERROR:SCHEMASV:SCHEMAV_CVC_COMPLEX_TYPE_3_2_1: Element 'when', attribute 'attribute': The attribute 'attribute' is not allowed.
/tmp/tmpKBNiWQ:46:0:ERROR:SCHEMASV:SCHEMAV_CVC_COMPLEX_TYPE_3_2_1: Element 'when', attribute 'input_dataset': The attribute 'input_dataset' is not allowed.
/tmp/tmpKBNiWQ:46:0:ERROR:SCHEMASV:SCHEMAV_CVC_COMPLEX_TYPE_4: Element 'when': The attribute 'input' is required but missing.
/tmp/tmpKBNiWQ:47:0:ERROR:SCHEMASV:SCHEMAV_CVC_COMPLEX_TYPE_3_2_1: Element 'when', attribute 'attribute': The attribute 'attribute' is not allowed.
/tmp/tmpKBNiWQ:47:0:ERROR:SCHEMASV:SCHEMAV_CVC_COMPLEX_TYPE_3_2_1: Element 'when', attribute 'input_dataset': The attribute 'input_dataset' is not allowed.
/tmp/tmpKBNiWQ:47:0:ERROR:SCHEMASV:SCHEMAV_CVC_COMPLEX_TYPE_4: Element 'when': The attribute 'input' is required but missing.
/tmp/tmpKBNiWQ:48:0:ERROR:SCHEMASV:SCHEMAV_CVC_COMPLEX_TYPE_3_2_1: Element 'when', attribute 'attribute': The attribute 'attribute' is not allowed.
/tmp/tmpKBNiWQ:48:0:ERROR:SCHEMASV:SCHEMAV_CVC_COMPLEX_TYPE_3_2_1: Element 'when', attribute 'input_dataset': The attribute 'input_dataset' is not allowed.
/tmp/tmpKBNiWQ:48:0:ERROR:SCHEMASV:SCHEMAV_CVC_COMPLEX_TYPE_4: Element 'when': The attribute 'input' is required but missing.]
Failed linting
```

This is triggered by the following block of XML,

``` xml
<data name="output_pos" format="fastq" label="With matched ID">
    <!-- TODO - Replace this with format="input:input_fastq" if/when that works -->
    <change_format>
        <when input_dataset="input_fastq" attribute="extension" value="fastqsanger" format="fastqsanger" />
        <when input_dataset="input_fastq" attribute="extension" value="fastqsolexa" format="fastqsolexa" />
         <when input_dataset="input_fastq" attribute="extension" value="fastqillumina" format="fastqillumina" />
         <when input_dataset="input_fastq" attribute="extension" value="fastqcssanger" format="fastqcssanger" />
    </change_format>
    <filter>output_choice_cond["output_choice"] != "neg"</filter>
</data>
```

This was written back in 2011, https://github.com/peterjc/pico_galaxy/commits/37598ac98aab731d3fe80d12e399e0d4dd5e92fb/tools/fastq/fastq_filter_by_id.xml

Is it possible to declare deprecated attributes in the XSD?

I have in this case simply switched to a more recent idiom: https://github.com/peterjc/pico_galaxy/commit/bab0d9113236ac6487cfe03db4c64ccb2e46104b

### PR DESCRIPTION
**Pull request by jmchilton**
Fixes #3552.

### CODE REVIEW

File: lib/galaxy/tools/xsd/galaxy.xsd [hunk 1]
```diff
@@ -4888,9 +4888,9 @@ for more examples.
 ]]></xs:documentation>
     </xs:annotation>
     <xs:sequence/>
-    <xs:attribute name="input" type="xs:string" use="required">
+    <xs:attribute name="input" type="xs:string">
       <xs:annotation>
-        <xs:documentation xml:lang="en">This value must be the attribute name of
+        <xs:documentation xml:lang="en">This attribute should be the name of
 the desired input parameter (e.g. ``input="out_format"`` above).</xs:documentation>
```
Comment (nsoranzo) [inline lib/galaxy/tools/xsd/galaxy.xsd:4894]: I think the documentation of `value` below is not correct, it says: `This value must also be an attribute name of an input parameter (e.g. ``value="interval"`` above).`, but should be: ```This must be a possible value of the ``input`` parameter (e.g. ``value="interval"`` above), or of the ``input_dataset``'s ``attribute.```

**Review by nsoranzo — APPROVED** [review]

**Comment by martenson** [comment]
```
test/functional/tools/output_format.xml
not ok 85 test/functional/tools/output_format.xml
    Warning: program compiled against libxml 208 using older 207
    -:22: element when: Schemas validity error : Element 'when': The attribute 'attribute' is required but missing.
    -:22: element when: Schemas validity error : Element 'when': The attribute 'input_dataset' is required but missing.
    -:23: element when: Schemas validity error : Element 'when': The attribute 'attribute' is required but missing.
    -:23: element when: Schemas validity error : Element 'when': The attribute 'input_dataset' is required but missing.
    - fails to validate
```

**Comment by nsoranzo** [comment]
The `validate-test-tools` build fail is legit, `use="required"` must be removed from `attribute` and `input_dataset` .

Also I think I'd like to have their definition order swapped, first `input_dataset` then `attribute`.

**Comment by jmchilton** [comment]
I believe I have addressed the relevant issues and this is ready to go.

**Pull request merged**